src/app/DIAmeterApplication.h

```diff
@@ -57,31 +69,82 @@ class DIAmeterApplication : public CruxApplication {
 	const ProteinVec& proteins, ///< proteins corresponding with peptides
 	const vector<const pb::AuxLocation*>& locations,  ///< auxiliary locations
 	TideMatchSet* matches, ///< object to manage PSMs
-	map<int, pair<double*, double*>>* ms1scan_intensity_rank_map,
+	ObservedPeakSet* observed,
+	// map<int, pair<double*, double*>>* ms1scan_intensity_rank_map,
+	map<int, boost::tuple<double*, double*, double*, int>>* ms1scan_mz_intensity_rank_map,
+	map<int, boost::tuple<double, double>>* ms1scan_slope_intercept_map,
+	// map<int, double*>* ms2scan_intensity_map,
+	map<int, boost::tuple<double*, double*, int>>* ms2scan_mz_intensity_map,
 	map<string, double>* peptide_predrt_map
   );
 
-  void computePrecIntRank(
+  void computePrecIntRankOld(
```

Comment:
There should be a comment describing why we have an "old" and "new"

Reply:
Actually - does this have to be 2 different functions? Could you merge into one function with a flag?